drv/i2c-devices/src/mwocp68.rs

```diff
@@ -62,6 +115,43 @@ impl From<pmbus::Error> for Error {
     }
 }
 
+// const MWOCP68_MFR_ID: &str = "Murata-PS";
+// const MWOCP68_MFR_MODEL: &str = "MWOCP68-3600-D-RM";
+
+const MWOCP68_REVISION_LEN: usize = 14;
+const MWOCP68_REVISION_FORMAT: &[u8; MWOCP68_REVISION_LEN] = b"XXXX-YYYY-0000";
+
+const MWOCP68_BLOCK_LENGTH: usize = 32;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum UpdateState {
+    WroteBootLoaderKey,
+    WroteProductKey,
+    BootedBootLoader,
+    StartedProgramming,
+    WroteBlock { offset: usize, checksum: u64 },
+    WroteLastBlock { checksum: u64 },
+    SentChecksum,
+    VerifiedChecksum,
+    RebootedPSU,
+    UpdateSuccessful,
+}
+
+impl UpdateState {
+    fn delay_ms(&self) -> u64 {
+        match self {
+            Self::WroteBootLoaderKey | Self::WroteProductKey => 3_000,
+            Self::BootedBootLoader => 1_000,
+            Self::StartedProgramming => 2_000,
+            Self::WroteBlock { .. } | Self::WroteLastBlock { .. } => 100,
+            Self::SentChecksum => 2_000,
```

Comment:
suuuper pedantic, feel free to ignore me: for other variants that have the same delay, we use `|` to combine them into the same match arm, but we don't for `StartedProgramming` and `SentChecksum`; for consistency's sake, perhaps:
```suggestion
            Self::StartedProgramming | Self::SentChecksum => 2_000,
            Self::WroteBlock { .. } | Self::WroteLastBlock { .. } => 100,
```

(I will freely admit that this is a goofy thing to care about, which is why i should clarify that I don't actually care.)

Reply:
I actually went the other way on this, and broke everything (except for the two `WroteBlock` variants) out as their own delay.